### PR TITLE
feat: migration for 0.15.x

### PR DESCRIPTION
**Pull request by TorstenDittmann**

### CODE REVIEW

File: src/Appwrite/Migration/Version/V14.php [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+            $this->projectDB->setNamespace("_{$this->project->getInternalId()}");
+
+            switch ($id) {
+                case 'attributes':
+                case 'indexes':
+                    try {
+                        /**
+                         * Create 'databaseInternalId' attribute
+                         */
+                        $this->createAttributeFromCollection($this->projectDB, $id, 'databaseInternalId');
```
Comment (lohanidamodar) [inline src/Appwrite/Migration/Version/V14.php:180]: `databaseId` attribute is missing?

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Hey I think few things were missing.

File: src/Appwrite/Migration/Version/V14.php [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+     */
+    protected function migrateCustomCollections(): void
+    {
+        try {
+            $this->pdo->prepare("ALTER TABLE IF EXISTS `{$this->projectDB->getDefaultDatabase()}`.`_{$this->project->getInternalId()}_collections` RENAME TO `_{$this->project->getInternalId()}_database_1`")->execute();
+        } catch (\Throwable $th) {
+            Console::warning($th->getMessage());
+        }
+        try {
+            $this->pdo->prepare("ALTER TABLE IF EXISTS `{$this->projectDB->getDefaultDatabase()}`.`_{$this->project->getInternalId()}_collections_perms` RENAME TO `_{$this->project->getInternalId()}_database_1_perms`")->execute();
```
[the remark this answers is not in the document]
Comment (TorstenDittmann) [reply src/Appwrite/Migration/Version/V14.php:75]: This got changed with internal ids 😉

File: src/Appwrite/Migration/Version/V14.php [hunk 3]
```diff
@@ -0,0 +1,685 @@
...
+            ")->execute();
+        } catch (\Throwable $th) {
+            Console::warning($th->getMessage());
+        }
+
+        try {
+            /**
+             * Add Database Internal ID for Collections.
+             */
+            $this->createAttributeFromCollection($this->projectDB, 'database_1', 'databaseInternalId', 'collections');
```
[the remark this answers is not in the document]
Comment (TorstenDittmann) [reply src/Appwrite/Migration/Version/V14.php:98]: good catch 👍🏻

File: src/Appwrite/Migration/Version/V14.php (hunk 1 above)
Comment (TorstenDittmann) [reply src/Appwrite/Migration/Version/V14.php:180]: Good catch 👍🏻

**Pull request merged**